### ISSUE TITLE
Support server grouping

### ISSUE THREAD

**Issue by corydeppen** [kendo-labs/dlinq-helpers#2]
Considering the Kendo UI DataSource supports [serverGrouping](http://docs.kendoui.com/api/framework/datasource#configuration-serverGrouping), it would be beneficial if the dynamic GroupBy extension method could be used with the incoming group parameter to perform the appropriate grouping.

**Comment by burkeholland** [comment]
Agreed. Looking into it.

**Comment by burkeholland** [comment]
Welllll.....I tried to do this with the DyanmicLinq library, and it's not so easy since you can't get a typed queryable from `GroupBy`.  Any ideas?  Pull requests accepted you know... :)

**Comment by gramx** [comment]
Interesting, a new passable list of column names could be created and then processed using the SQL/LINQ 'group by'? Maybe something like this example from Jon Skeet for an IEnumerable query.

```
 persons.GroupBy(p => p.PersonId, p => p.car,
                          (key, g) => new { PersonId = key, Cars = g.ToList() });
```

**Comment by burkeholland** [comment]
Interesting idea, but I'm still unclear on the implementation

**Comment by gramx** [comment]
What part of the implementation are you referring to?

```
 1. The view/JS gathering the grid's grouped data.*
 2. The view/JS sending the grouped data to the controller.
 3. The JSON group data object structure (transit & serialization)
 4. The controller setting up a DB query with group by?
```

I may end up looking at this in more detail in a few months and depending on the direction of our project takes.
*Would this require modifications of the control?

Edit: Our project went a different direction with the open release of the newer Kendo controls. We are reviewing a larger scope.

**Comment by nlips** [comment]
The .QueryByCube() function provided by the AdaptiveLINQ component (www.adaptivelinq.com) can help you to implement an OData feed performing server side analysis query (GroupBy + Aggregation).
For example, defining a two dimensions cube: customer and product category, you can easily obtain the total sales: 
- by customer: salesAnalysis?$select=customer,sales
- by product category: salesAnalysis?$select=productCategory,sales
- by cusytomer and product category: salesAnalysis?$select=customer,productCategory,sales

**Comment by burkeholland** [comment]
Does that help only in terms of OData?

**Comment by burkeholland** [comment]
@gramx Good luck!

**Comment by gramx** [comment]
@burkeholland Thanks, the team is looking forward to some of the new open items. :+1:

**Comment by kpko** [comment]
Hi, is there any progress on this issue? I think this one is crucial for Kendo UI when using server side actions. When I use server side paging, there is no use in using client side grouping. I currently use a hack in our application leveraging the Kendo UI MVC Helpers (setting the transport of the Kendo UI DataSource to aspnetmvc and using the corresponding libraries on the server side.) 

Telerik has already implemented this functionality. Grouping works as expected. Shouldn't it be easy to implement this in the dlinq-helper library? So we wouldn't have to use the Kendo.MVC library in our single page applications at all.

**Comment by CraigHead** [comment]
I would still love to see this implemented. I took a look at implementing my own version but as you said before... it got complicated. 👍

**Comment by mshtawythug** [comment]
@CraigHead @kpko @burkeholland @gramx  
I added a fork with a bunch of fixes one of them is server side grouping check it out 
https://github.com/mshtawythug/dlinq-helpers/tree/master/Kendo.DynamicLinq

**Comment by HemenRohani** [comment]
I add fork to add ServerGrouping for one field and fix VS2015 (non-static method requires a target) error.
[link to fork](https://github.com/HemenRohani/dlinq-helpers)

**Comment by crazyfx1** [comment]
Is any of those forks which implement serverGrouping is going to be pulled back to the master?

**Comment by IAfanasov** [comment]
@HemenRohani @mshtawythug why don't you make pull requests?
people need this feature! (:

**Comment by CraigHead** [comment]
@HemenRohani @mshtawythug @burkeholland Is your fork up-to-date?

**Comment by IAfanasov** [comment]
@CraigHead I tried fork of @mshtawythug. There is some easy to fix bug with grouping and no support for aggregates and it works. We are going to use it in production.

@HemenRohani made only some changes to readme.md file

I see several pull requests waiting for approval for a long time. Looks like @kendo-labs no longer cares about this repo

**Comment by mshtawythug** [comment]
@CraigHead yes it is, it is being  used for production data in my company, The grouping still needs some work on aggregates but everything else works like a charm

**Comment by HemenRohani** [comment]
@CraigHead yes it is, server grouping for one level with aggregates is working.
